packages/frontend/core/src/blocksuite/ai/components/chat-action-list.ts

```diff
@@ -89,11 +107,48 @@ export class ChatActionList extends LitElement {
   accessor messageId: string | undefined = undefined;
 
   @property({ attribute: false })
-  accessor layoutDirection: 'horizontal' | 'vertical' = 'vertical'; // New property for layout direction
+  accessor layoutDirection: 'horizontal' | 'vertical' = 'vertical';
 
   @property({ attribute: false })
   accessor withMargin = false;
 
+  override connectedCallback() {
+    super.connectedCallback();
+    this._disposables.add(
+      this._gfx.selection.slots.updated.on(() => {
+        this._debouncedUpdate();
+      })
+    );
+    this._disposables.add(
+      this.host.selection.slots.changed.on(() => {
+        this._debouncedUpdate();
+      })
+    );
+  }
+
+  override disconnectedCallback() {
+    super.disconnectedCallback();
+    this._debouncedUpdate.cancel();
+    this._disposables.dispose();
```

Comment:
I think you don't need to add the `_disposables` field and call `dispose()` if you've already extended `WithDisposable`.